src/main/histFileMod.F90

```diff
@@ -318,6 +332,93 @@ subroutine hist_printflds()
        call shr_sys_flush(iulog)
     end if
 
+    ! Print master field list in separate text file when namelist
+    ! variable requests it. Text file is formatted in the .rst
+    ! (reStructuredText) format for easy introduction of the file to
+    ! the CTSM's web-based documentation.
+    if (masterproc .and. hist_master_list_file) then
+       ! Hardwired table column widths to fit the table on a computer
+       ! screen. Some strings will be truncated as a result of the
+       ! current choices (4, 39, 94, 65). In sphinx (ie the web-based
+       ! documentation), text that has not been truncated will wrap
+       ! around in the available space.
+       width_col_1 = 4  ! column that shows the variable number, nf
+       width_col_2 = 39  ! variable name column
+       width_col_3 = 94  ! long description column
+       width_col_4 = 65  ! units column
+       width_col_sum = width_col_1 + width_col_2 +  &  ! sum of widths plus
+                       width_col_3 + width_col_4 + 3  ! ...3 blank spaces
+
+       ! Convert integer widths to strings for use in format statements
+       ! These write statements are not outputting to files
+       write(str_w_col_1,'(i0)') width_col_1
+       write(str_w_col_2,'(i0)') width_col_2
+       write(str_w_col_3,'(i0)') width_col_3
+       write(str_w_col_4,'(i0)') width_col_4
+       write(str_w_col_sum,'(i0)') width_col_sum
+
+       ! Open master_list_file
+       master_list_file = getavu()  ! get next available file unit number
+       open(unit = master_list_file, file = 'master_list_file.rst',  &
+            status = 'new', action = 'write', form = 'formatted')
+
+       ! File title
+       fmt_txt = '(19a)'
```

Comment:
I'm pretty sure you can just use '(a)' to avoid needing to count characters. (Not critical to fix.)